### PR TITLE
Combine `RUN` commands to reduce rocm image size

### PR DESCRIPTION
**Pull request by caponetto**
## Description
Image size comparison before -> after these changes:

- `rocm-jupyter-tensorflow-ubi9-python-3.9`: 38 GB -> 35 GB
- `rocm-jupyter-pytorch-ubi9-python-3.9`: 60 GB -> 46 GB

Note: Both files are now exactly the same, except for the labels.

## How Has This Been Tested?

Run the build workflow with the changes included

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: jupyter/rocm/pytorch/ubi9-python-3.9/Dockerfile [hunk 1]
```diff
@@ -14,14 +14,10 @@ LABEL name="odh-notebook-jupyter-rocm-pytorch-ubi9-python-3.9" \
...
-    rm -f ./Pipfile.lock && \
-    # Replace Notebook's launcher, "(ipykernel)" with Python's version 3.x.y \
+RUN echo "Installing softwares and packages" && micropipenv install && rm -f ./Pipfile.lock && \
+    # Replace Notebook's launcher, "(ipykernel)" with Python's version 3.x.y
     sed -i -e "s/Python.*/$(python --version | cut -d '.' -f-2)\",/" /opt/app-root/share/jupyter/kernels/python3/kernel.json && \
-    # Disable announcement plugin of jupyterlab \
-    jupyter labextension disable "@jupyterlab/apputils-extension:announcements"
-
-# Fix permissions to support pip in Openshift environments
-RUN chmod -R g+w /opt/app-root/lib/python3.9/site-packages && \
```
Comment (jiridanek) [inline jupyter/rocm/pytorch/ubi9-python-3.9/Dockerfile:26]: * https://github.com/opendatahub-io/notebooks/issues/539

Comment (jiridanek) [reply jupyter/rocm/pytorch/ubi9-python-3.9/Dockerfile:26]: I like this, I think this should be done in all images that use fix-permissions tool/script!

**Review by jiridanek — APPROVED** [review]

**Comment by jiridanek** [comment]
/override ci/prow/rocm-runtimes-ubi9-e2e-tests
/override ci/prow/runtime-rocm-pytorch-ubi9-python-3-9-pr-image-mirror
/override ci/prow/images
/override ci/prow/rocm-notebooks-e2e-tests

these things don't build because image is too big, and this PR will shrink the image

**Comment by jiridanek** [comment]
/override ci/prow/runtimes-ubi8-e2e-tests
/override ci/prow/runtimes-ubi9-e2e-tests
/override ci/prow/runtime-rocm-tensorflow-ubi9-python-3-9-pr-image-mirror
/override ci/prow/notebook-rocm-jupyter-pyt-ubi9-python-3-9-pr-image-mirror
/override ci/prow/notebook-rocm-jupyter-tf-ubi9-python-3-9-pr-image-mirror
